builder.go

```diff
@@ -2,15 +2,16 @@ package wazero
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/tetratelabs/wazero/api"
-	"github.com/tetratelabs/wazero/internal/leb128"
-	"github.com/tetratelabs/wazero/internal/u64"
 	"github.com/tetratelabs/wazero/internal/wasm"
 )
 
-// ModuleBuilder is a way to define a WebAssembly Module in Go.
+// HostModuleBuilder is a way to define host functions (in Go), so that a
+// WebAssembly binary (ex. %.wasm file) can import and use them.
+//
+// Specifically, this implements the host side of an Application Binary
+// Interface (ABI) like as WASI or AssemblyScript.
```

Comment:
```suggestion
// Interface (ABI) like WASI or AssemblyScript.
```

Also not sure I'd mention a language (AssemblyScript) in this doc